applications/FluidDynamicsApplication/custom_processes/weighted_divergence_calculation_process.h

```diff
@@ -0,0 +1,338 @@
+
+                    // Compute divergence and velocity seminorm
+                    const double aux_current_divergence = grad_x[0] + grad_y[1] + grad_z[2];
+                    const double aux_current_velocity_seminorm = grad_x[0]*grad_x[0] + grad_x[1]*grad_x[1] + grad_x[2]*grad_x[2] + grad_y[0]*grad_y[0] + grad_y[1]*grad_y[1] + grad_y[2]*grad_y[2] + grad_z[0]*grad_z[0] + grad_z[1]*grad_z[1] + grad_z[2]*grad_z[2];
+                    const double gauss_point_volume = r_integration_points[point_number].Weight() * detJ0;
+                    divergence_current += std::pow(aux_current_divergence,2) * gauss_point_volume;
+                    velocity_seminorm_current += aux_current_velocity_seminorm * gauss_point_volume;
+                }
+                // Retrieve divergence from previous time step
+                auto divergence_old = it_elem->GetValue(DIVERGENCE);
+                auto velocity_seminorm_old = it_elem->GetValue(VELOCITY_H1SEMINORM);
+
+                // Compute divergence weighted time average
+                auto divergence_current_avg = std::sqrt(((time_step_previous-time_coefficient*final_time) * std::pow(divergence_old,2) + (time_step_current - time_step_previous) * divergence_current) /  (time_step_current-time_coefficient*final_time));
+                it_elem->SetValue(DIVERGENCE,divergence_current_avg);
+
+                // Compute divergence_norm weighted time average
+                auto velocity_seminorm_current_avg = std::sqrt(((time_step_previous-time_coefficient*final_time) * std::pow(velocity_seminorm_old,2) + (time_step_current - time_step_previous) * velocity_seminorm_current) /  (time_step_current-time_coefficient*final_time));
+                it_elem->SetValue(VELOCITY_H1SEMINORM,velocity_seminorm_current_avg);
+
+            }
+        }
+
+        KRATOS_CATCH("");
+    }
+
+
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "WeightedDivergenceCalculationProcess";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "WeightedDivergenceCalculationProcess";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+        this->PrintInfo(rOStream);
+    }
+
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+
+    ///@}
+
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ModelPart& mrModelPart;
+
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+
+    ///@}
+    ///@name Un accessible methods
+    ///@{
+
+    /// Assignment operator.
+    WeightedDivergenceCalculationProcess& operator=(WeightedDivergenceCalculationProcess const& rOther);
```

Comment:
```suggestion
    WeightedDivergenceCalculationProcess& operator=(WeightedDivergenceCalculationProcess const& rOther) = delete;
```
And move it to the life cycle section in the top of the file.